tree: mark whether OPERATOR is used in UnaryExpr

otan opened the pull request:
Seems like there's some weird parenthesising thing happening
with OPERATOR such that it is worth storing whether OPERATOR
was used in tree.UnaryExpr.

This commit changes UnaryOperator to be composed of UnaryOperatorSymbol
(the former usage of UnaryOperator) and IsOperator to do so.

Resolves #65078 - not sure whether we need to do Comparison/BinaryExpr.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65107)
<!-- Reviewable:end -->

otan committed:
tree: mark whether OPERATOR is used in UnaryExpr

Seems like there's some weird parenthesising thing happening
with OPERATOR such that it is worth storing whether OPERATOR
was used in tree.UnaryExpr.

This commit changes UnaryOperator to be composed of UnaryOperatorSymbol
(the former usage of UnaryOperator) and IsOperator to do so.

Release note: None

rafiss reviewed (approved):
ty!

otan commented:
bors r=rafiss
